SIFT custom

carpino opened the pull request:
Hello, 

ive crearted a procedure to instantiate a SIFT object with custom values, 

ive created a new procedure leaving untouched the default unparametrized procedure,

hope this will be included in the dev branch,

(ive closed the pull request just created in the wrong branch)

@deadprogram

carpino committed:
SIFT custom

deadprogram commented on features2d.go:
```diff
@@ -943,6 +943,39 @@ func NewSIFT() SIFT {
 	return SIFT{p: unsafe.Pointer(C.SIFT_Create())}
 }
 
+func NewSIFTCustom(nfeatures *int, nOctaveLayers *int, contrastThreshold *float64, edgeThreshold *float64, sigma *float64) SIFT {
```
```suggestion
func NewSIFTWithParams(nfeatures *int, nOctaveLayers *int, contrastThreshold *float64, edgeThreshold *float64, sigma *float64) SIFT {
```

deadprogram commented on features2d.go:
```diff
@@ -943,6 +943,39 @@ func NewSIFT() SIFT {
+		numSigma = *sigma
+	}
+
+	return SIFT{p: unsafe.Pointer(C.SIFT_CreateCustom(C.int(numFeatures), C.int(numOctaveLayers), C.double(numContrastThreshold), C.double(numEdgeThreshold), C.double(numSigma)))}
```
```suggestion
	return SIFT{p: unsafe.Pointer(C.SIFT_CreateWithParams(C.int(numFeatures), C.int(numOctaveLayers), C.double(numContrastThreshold), C.double(numEdgeThreshold), C.double(numSigma)))}
```

deadprogram commented on features2d.h:
```diff
@@ -97,6 +97,7 @@ struct MultiDMatches FlannBasedMatcher_KnnMatch(FlannBasedMatcher f, Mat query,
 void DrawKeyPoints(Mat src, struct KeyPoints kp, Mat dst, const Scalar s, int flags);
 
 SIFT SIFT_Create();
+SIFT SIFT_CreateCustom(int nfeatures, int nOctaveLayers, double contrastThreshold, double edgeThreshold, double sigma);
```
```suggestion
SIFT SIFT_CreateWithParams(int nfeatures, int nOctaveLayers, double contrastThreshold, double edgeThreshold, double sigma);
```

deadprogram commented on features2d.cpp:
```diff
@@ -590,6 +590,11 @@ SIFT SIFT_Create() {
     return new cv::Ptr<cv::SIFT>(cv::SIFT::create());
 }
 
+SIFT SIFT_CreateCustom(int nfeatures, int nOctaveLayers, double contrastThreshold, double edgeThreshold, double sigma) {
```
```suggestion
SIFT SIFT_CreateWithParams(int nfeatures, int nOctaveLayers, double contrastThreshold, double edgeThreshold, double sigma) {
```

deadprogram commented:
Hello @carpino please see my suggested changes inline.

See https://github.com/hybridgroup/gocv/blob/release/CONTRIBUTING.md#how-to-add-a-function-from-opencv-to-gocv for some explanation.

Also, can you please add a smoke test as described in the contribution guidelines?

Thank you!

carpino committed:
SIFT_WIthParams and smoke test for NewSIFTWithParams

carpino commented:
> Hello @carpino please see my suggested changes inline.
> 
> See https://github.com/hybridgroup/gocv/blob/release/CONTRIBUTING.md#how-to-add-a-function-from-opencv-to-gocv for some explanation.
> 
> Also, can you please add a smoke test as described in the contribution guidelines?
> 
> Thank you!

Hi @deadprogram, thanks for the quick response, changed the name according to your suggestion and contribution guidelines,

added smoke test that follow the structure of the test for SIFT without params, checking the correct number of features, generally I should check for a number of features not greater than the params but for the image used in the test SIFT extract an higher number of features if created without params, so I check an inequality

deadprogram commented:
Thanks for the addition @carpino now squash/merging.
